ui/hooks/useOriginTrustSignals.ts

```diff
@@ -0,0 +1,47 @@
+import { RecommendedAction } from '@metamask/phishing-controller';
+import { useSelector } from 'react-redux';
+import { getUrlScanCacheResult } from '../selectors/selectors';
+import { TrustSignalDisplayState, TrustSignalResult } from './useTrustSignals';
+
+type UrlScanCacheResult = {
+  result: {
+    domainName: string;
+    recommendedAction: string; // 'NONE', 'WARN', 'BLOCK'
+  };
+  timestamp: number;
+};
+
+function getTrustState(
+  cachedResult: UrlScanCacheResult | undefined,
+): TrustSignalDisplayState {
+  const recommendedAction = cachedResult?.result?.recommendedAction;
+
+  if (!recommendedAction) {
+    return TrustSignalDisplayState.Unknown;
+  }
+
+  switch (recommendedAction) {
+    case RecommendedAction.Block:
+      return TrustSignalDisplayState.Malicious;
+    case RecommendedAction.Warn:
+      return TrustSignalDisplayState.Warning;
+    case RecommendedAction.Verified:
+      return TrustSignalDisplayState.Verified;
+    case RecommendedAction.None:
+    default:
+      return TrustSignalDisplayState.Unknown;
+  }
+}
+
+export function useOriginTrustSignals(origin: string): TrustSignalResult {
+  const urlScanCacheResult = useSelector((state) =>
+    getUrlScanCacheResult(state, origin),
+  );
```

Comment:
Why are adding an inline selector here? I think that this will have an impact as we will recompute the value on every component render. Also we are not leveraging any memoizing techniques that we have available (like the createDeepEqualSelector).

Reply:
Nevermind my comment... we are passing a prop into the selector, so we need the arrow function here.